app/esper/rekall_vgrid.py

```diff
@@ -0,0 +1,354 @@
+"""Utilities for building VGrid inputs from Rekall interval sets
+
+VGrid displays a list of VBlocks. Each VBlock can have multiple tracks, and
+each track shows a timeline of intervals in that track, as well as drawing
+objects and showing metadata of the interval as configured.
+
+There are two main use cases of VGrid:
+    1. VideoVBlocks. This mode displays a VBlock for each video. Each track
+        in the VBlock is an IntervalSet to visualize in that video.
+    2. IntervalVBlocks. This mode displays a VBlock for each interval/clip.
+        Each track in the VBlock is an IntervalSet to visualize in that
+        interval/clip.
+
+The VBlocksBuilders in this module helps building the JSON structure for this
+list of VBlocks.
+
+Example of using VideoVBlock:
+
+    # face_collection is a DomainIntervalCollection of face intervals.
+
+    # We want to see a list of VBlocks where each is a video in the collection
+    # We want to see two tracks in each VBlock: one showing all faces in the
+    # collection with bounding boxes drawn and a flag metadata set; another
+    # showing all captions of the video.
+
+    json = VideoVBlocksBuilder()\\
+        .add_track(
+            VideoTrackBuilder('faces', face_collection)\\
+                .set_draw_type(DrawType_Bbox())\\
+                .add_metadata('flag', Metadata_Flag()))\\
```

Comment:
.add_metadata should probably be a function of the payload somehow? e.g. adding metadata might be conditional, like add a flag if this caption text == "foobar" or if this gender is female or whatever

Reply:
There are two ways to express this in the proposed module:
1. Metadata_Flag class contains the state of "flag" or "no flag" in its JSON value.
2. Have a ConditionalMetadata class with c'tor `(predicate, delegate_metadata)`. We can either handle null metadata here or in VGrid.

Reply:
Went with option 1 in `rekall.vgrid_utils.vblocks_builder`.